test/test_collection.py

```diff
@@ -26,3 +50,80 @@ def testCollectionsCount(server, userIdx, expected, collections, users):
     resp = server.request(path='/collection/details', user=users[userIdx])
     assertStatusOk(resp)
     assert resp.json['nCollections'] == expected
+
+
+def testSingleCollectionMetaExists(server, collection, admin):
+    resp = server.request(path='/collection/%s' % collection['_id'], user=admin)
+    assertStatusOk(resp)
+    assert 'meta' in resp.json
+
+
+def testListCollectionMetaExists(server, collection, admin):
+    resp = server.request(path='/collection', user=admin)
+    assertStatusOk(resp)
+    assert all(('meta' in x) for x in resp.json)
+
+
+def testCollectionSetMetadata(server, collection, metadata, admin):
+    resp = server.request(
+        path='/collection/%s/metadata' % collection['_id'],
+        user=admin,
+        method='PUT',
+        body=json.dumps(metadata),
+        type='application/json')
+
+    assertStatusOk(resp)
+    assert resp.json['meta'] == metadata
+
+    # Check that fetching the object again yields the same result
+    newDoc = server.request(
+        path='/collection/%s' % collection['_id'],
+        user=admin,
+        method='GET')
+
+    assert newDoc.json['meta'] == metadata
+
+
+def testCollectionDeleteMetadata(server, collectionWithMeta, metadata, admin):
+    collection = collectionWithMeta(metadata)
+    resp = server.request(
+        path='/collection/%s/metadata' % collection['_id'],
+        user=admin,
+        method='DELETE',
+        body=json.dumps(list(metadata.keys())),
+        type='application/json')
+    assertStatusOk(resp)
+    assert resp.json['meta'] != metadata
+    assert resp.json['meta'] == {}
+
+    newDoc = server.request(
+        path='/collection/%s' % collection['_id'],
+        user=admin,
+        method='GET')
+    assert newDoc.json['meta'] != metadata
+    assert newDoc.json['meta'] == {}
+
+
```

Comment:
I think it's worthwhile to add 1-2 more tests at the HTTP API layer to ensure that old collections with no `meta` field (which you'll have to create artificiality) have it appropriately added to the `GET /collection` and `GET /collection/:id` endpoints. This behavior will be very important to users.